fleet_adapter_r3/fleet_adapter_r3/RobotCommandHandle.py

```diff
@@ -220,6 +225,23 @@ def follow_new_path(
         self._quit_dock_event.clear()
 
         self.node.get_logger().info("Received new path to follow...")
+
+        if self.use_safe_nav:
+            name_waypoints = []
+            # Discard first waypoint due to non-match with existing waypoints.
+            waypoints = waypoints[1:]
```

Comment:
non-blocking: why do we need to discard the first waypoint?

Reply:
Based on tests, RMF Traffic Scheduler often dispatch with an array of waypoints, **typically starting with ones that do not share the same coordinates as any waypoints as defined on a user's RMF Map**.

Still unsure why the RMF Traffic Scheduler does that but this is a workaround to ensure that waypoints received are waypoints that exists on the RMF Map. 

Additionally, RMF Traffic Scheduler also often send duplicate waypoints as part of the array of waypoints during dispatch.

Reply:
Hmm I see ok!